sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -0,0 +1,459 @@
+        this.vaultUrl,
+        roleScope,
+        continuationState.continuationToken,
+        options
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      if (currentSetResponse.value) {
+        yield currentSetResponse.value.map(mappings.roleAssignment.generatedToPublic, this);
+      } else {
+        break;
+      }
+    }
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the iteration of all the available results of {@link listRoleAssignments}.
+   * @param {string} roleScope The scope of the role assignments.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleAssignmentsAll(
+    roleScope: RoleAssignmentScope,
+    options?: ListRoleAssignmentsOptions
+  ): AsyncIterableIterator<KeyVaultRoleAssignment> {
+    const f = {};
+
+    for await (const page of this.listRoleAssignmentsPage(roleScope, f, options)) {
+      for (const item of page) {
+        yield item;
+      }
+    }
+  }
+
+  /**
+   * Iterates over all of the available role assignments in an Azure Key Vault.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new KeyVaultAccessControlClient(url, credentials);
+   * for await (const roleAssignment of client.listRoleAssignments("/")) {
+   *   console.log("Role assignment: ", roleAssignment);
+   * }
+   * ```
+   * @summary Lists all of the role assignments in a given scope.
+   * @param {string} roleScope The scope of the role assignments.
+   * @param {ListRoleAssignmentsOptions} [options] The optional parameters.
+   */
+  public listRoleAssignments(
+    roleScope: RoleAssignmentScope,
+    options: ListRoleAssignmentsOptions = {}
+  ): PagedAsyncIterableIterator<KeyVaultRoleAssignment> {
+    const requestOptions = operationOptionsToRequestOptionsBase(options);
+    const span = createSpan("listRoleAssignments", requestOptions);
+    const updatedOptions: ListRoleAssignmentsOptions = {
+      ...requestOptions,
+      ...setParentSpan(span, requestOptions)
+    };
+
+    const iter = this.listRoleAssignmentsAll(roleScope, updatedOptions);
+
+    span.end();
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) =>
+        this.listRoleAssignmentsPage(roleScope, settings, updatedOptions)
+    };
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the pagination of {@link listRoleDefinitions}.
+   * @param {string} roleScope The scope of the role definition.
+   * @param {PageSettings} continuationState An object that indicates the position of the paginated request.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleDefinitionsPage(
+    roleScope: RoleAssignmentScope,
+    continuationState: PageSettings,
```

Comment:
Should we use a custom type here instead of `PageSettings` since maxPageSize isn't supported? I think it feels weird to expose configuration options we ignore.

We did this for search: https://github.com/Azure/azure-sdk-for-js/blob/80c7724c1ecdae1efac87eb2bd78bbfa035f3f64/sdk/search/search-documents/src/indexModels.ts#L76